web/app/themes/clarity/inc/maintenance.php

```diff
@@ -0,0 +1,398 @@
+    public function options_page(): void
+    {
+        add_submenu_page(
+            'options-general.php', /* Parent Menu Slug */
+            'Maintenance settings', /* Page Title */
+            'Maintenance', /* Menu Title */
+            $this->capability, /* Capability */
+            'maintenance', /* Menu Slug */
+            [$this, 'render_options_page'], /* Callback */
+        );
+    }
+
+    /**
+     * Renders the options page
+     */
+    public function render_options_page(): void
+    {
+
+        // check user capabilities
+        if (!current_user_can($this->capability)) {
+            return;
+        }
+
+        if (isset($_GET['settings-updated'])) {
+            add_settings_error('maintenance_messages', 'maintenance_message', __('Settings Saved', 'maintenance'), 'updated');
+        }
+
+        $options = get_option('maintenance_options');
+        $notification_title = $options['notification_title'] ?? '';
+        $notification_content = $options['notification_content'] ?? '';
+        $notification_level = $options['notification_level'] ?? 'info';
+
+        ?>
+        <div class="wrap">
+            <h1><?php echo esc_html(get_admin_page_title()); ?></h1>
+            <h2 class="description"></h2>
+            <form action="options.php" method="post">
+                <?php
+                settings_fields('maintenance');
+                do_settings_sections('maintenance');
+                ?>
+                <h3>Preview</h3>
+                <?php
+                // Display a preview of the notification banner
+                $this->show_notification_banner($notification_level, $notification_title, $notification_content, true);
+                submit_button('Save Settings');
+                ?>
+            </form>
+        </div>
+        <?php
+    }
+
+    /**
+     * Render a settings field.
+     *
+     * @param array $args A mixed array of arguments to render the different field types
+     */
+    public function render_field(array $args): void
+    {
+
+        $field = $args['field'];
+
+        // Get the value of the setting we've registered with register_setting()
+        $options = get_option('maintenance_options');
+        // Generate a unique key so that dismissed banners are reshown if the content changes
+        $notification_hash = $this->get_notification_hash($options['notification_content'], $options['notification_title']);
+
+        switch ($field['type']) {
+            case "text":
+            {
+                ?>
+                <input
+                    type="text"
+                    id="<?php echo esc_attr($field['id']); ?>"
+                    name="maintenance_options[<?php echo esc_attr($field['id']); ?>]"
+                    value="<?php echo isset($options[$field['id']]) ? esc_attr($options[$field['id']]) : ''; ?>"
+                >
+                <p class="description">
+                    <?php esc_html_e($field['description'], 'maintenance'); ?>
```

Comment:
```php
Warning: Undefined array key "description" in /bedrock/web/app/themes/clarity/inc/maintenance.php on line 237
```

Reply:
This has been updated to only access the description array key if a value is set.